Tiltfile

```diff
@@ -8,6 +8,8 @@ tools_bin = "./hack/tools/bin"
 
 #Add tools to path
 os.putenv("PATH", os.getenv("PATH") + ":" + tools_bin)
+os.putenv("EXP_KUBEADM_BOOTSTRAP_FORMAT_IGNITION", "true")
+os.putenv("FLATCAR_VERSION", "3374.2.1")
```

Comment:
why is this here and not with the other defaulted env variables? Even if we default these we should still allow them to be overwritten

Reply:
Thanks for feedback @CecileRobertMichon. Are you referring here only to `FLATCAR_VERSION` or also to `EXP_KUBEADM_BOOTSTRAP_FORMAT_IGNITION`? And what other env variables do you have in mind? As I said in the previous [comment](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2890#issuecomment-1386795155), I wasn't sure what's the best place to put those.

Perhaps `FLATCAR_VERSION` could be added to `substitutions` map or be deeply wired into Tilt settings.

`EXP_KUBEADM_BOOTSTRAP_FORMAT_IGNITION` may probably need to go in some different place, as it must be set when deploying the CAPI controlplane. Please advice me where should I put those so we can get this PR merged.

Thanks in advance.

(BTW, I didn't trigger the test or re-run them, as there is a temporary commit pushed which needs to be squashed and I know the tests are passing, as recently changed code only touches Tiltfile and I wanted to save some resources and notifications churn)

Reply:
I was referring to both.

- `FLATCAR_VERSION` should go in https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b438dbec0b0b5b26dc213a2d19ccbee1557fa57e/Tiltfile#L17. If we want it to be hardcoded (not user configuarable, follow `capi_version` as an example. If you do want it to be configurable, you can follow `kubernetes_version` as the example. This is what the dev experience for configuring it would look like: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/flavors/README.md#defining-variable-overrides
- `EXP_KUBEADM_BOOTSTRAP_FORMAT_IGNITION ` goes in https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/Makefile#L709 with the other exp variables

Reply:
Fixed, thank you for pointers.